sharktank/sharktank/utils/iree.py

```diff
@@ -753,3 +759,251 @@ def run_model_with_iree_run_module(
+
+
+def validate_and_get_irpa_path(request):
+    """
+    Validate and get IRPA path from pytest request configuration.
+
+    Args:
+        request: pytest request fixture
+
+    Returns:
+        str: Path to the IRPA file
+
+    Raises:
+        pytest.skip: If IRPA path is not provided or file doesn't exist
+    """
+    from pytest import skip
+
+    # Get IRPA path from command line argument
+    irpa_path = request.config.getoption("--parameters")
+
+    # Skip test if no IRPA path provided
+    if irpa_path is None:
+        skip("No IRPA path provided. Use --parameters to specify the IRPA file.")
+
+    # Skip test if IRPA file doesn't exist
+    if not Path(irpa_path).exists():
+        skip(f"IRPA file not found: {irpa_path}")
+
+    return irpa_path
+
+
+def run_iree_vs_torch_fx(
+    module: torch.nn.Module,
+    input_args=(),
+    kwargs=None,
+    *,
+    atol=1e-4,
+    rtol=0.0,
+    entrypoint="run_forward",
+    parameters_path=None,
+    compile_flags: list[str] | None = None,
+    driver="hip",
+    device_count=1,
+    directory=".",
+):
+    """
+    Wrapper for MLIR export via FxProgramsBuilder(model) and IREE vs Torch eager comparison.
+
+    Args:
+      module: torch.nn.Module under test
+      input_args: example positional inputs (tuple required)
+      kwargs: example kwargs
+      atol/rtol: tolerances passed to torch.testing.assert_close
+      entrypoint: the method name exported/invoked ("run_forward" by default)
+      parameters_path: Optional path to parameters file
+      compile_flags: List of compilation flags for iree
+      driver: IREE driver to use
+      device_count: Number of devices
+    """
+    with tempfile.TemporaryDirectory() as td:
+        td = Path(td)
+        mlir_path = td / "module.mlir"
+        vmfb_path = td / "module.vmfb"
+
+        # Export to MLIR and get torch reference
+        torch_output, _ = export_torch_module_to_mlir(
+            module=module,
+            input_args=input_args,
+            kwargs=kwargs,
+            mlir_path=mlir_path,
+            target_fn=entrypoint,
+        )
+
+        # Compile MLIR to VMFB
+        if compile_flags is None:
+            raise ValueError("compile_flags must be provided")
+
+        iree.compiler.compile_file(
+            str(mlir_path),
+            output_file=str(vmfb_path),
+            extra_args=compile_flags,
+        )
+        iree_devices = get_iree_devices(driver=driver, device_count=device_count)
+        # Run with IREE
+        iree_output = run_iree_module_from_vmfb(
+            vmfb_path=vmfb_path,
+            devices=iree_devices,
+            args=input_args,
+            entrypoint=entrypoint,
+            parameters_path=parameters_path,
+        )
+        # Compare outputs
+        compare_iree_torch_outputs(
+            iree_output=iree_output,
+            torch_output=torch_output,
+            atol=atol,
+            rtol=rtol,
+        )
+    del iree_devices
+    gc.collect()
```

Comment:
What kind of problems did you run into that requires manual GC?
Is it something related to the explanation in [with_iree_device_context](https://github.com/nod-ai/shark-ai/blob/d5c56eb0007799f39581413025fa5bc0f63fd707/sharktank/sharktank/utils/iree.py#L223).